modyn/trainer_server/internal/grpc/trainer_server_grpc_servicer.py

```diff
@@ -75,12 +100,39 @@ def start_training(
             return StartTrainingResponse(training_started=False)
 
         pretrained_model_path: Optional[pathlib.Path] = None
-        if request.pretrained_model_path is not None and request.pretrained_model_path != "":
-            pretrained_model_path = self._modyn_base_dir / pathlib.Path(request.pretrained_model_path)
-            if not pretrained_model_path.exists():
-                logger.error(f"Pretrained Model Path {pretrained_model_path} does not exist. Cannot start training.")
+        if request.use_pretrained_model and request.pretrained_model_id is not None:
+            fetch_request = FetchModelRequest(model_id=request.pretrained_model_id)
+            fetch_resp: FetchModelResponse = self.model_storage_stub.FetchModel(fetch_request)
+
+            if not fetch_resp.success:
+                logger.error(
+                    f"Pretrained Model {request.pretrained_model_id} cannot be fetched from model storage. "
+                    f"Training cannot be started."
+                )
                 return StartTrainingResponse(training_started=False)
 
+            last_progress = 0.0
+
+            def download_callback(current_progress: float) -> None:
+                nonlocal last_progress
+                for emit_perc in EMIT_MESSAGE_PERCENTAGES:
+                    if last_progress <= emit_perc < current_progress:
```

Comment:
sorry, but I actually don't know how to read this comparison statement haha. Can you add brackets to make this a bit more readable? That would be great :D I don't know in which order this is evaluated

Reply:
This feature of python is called comparison operator chaining and syntactic sugar for (in this case):
`last_progress <= emit_perc AND emit_perc < current_progress`
I agree, not the most readable solution if one is not used to it.
Should I still change this?

Reply:
ahh it's actually read like the mathematical statement. Thank you - I actually did not know this! No we can keep it that way.